cmd/snap/cmd_run.go

```diff
@@ -1186,7 +1186,13 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	// https://www.freedesktop.org/wiki/Software/systemd/ControlGroupInterface/
 	_, appName := snap.SplitSnapApp(snapApp)
 	needsTracking := true
-	if app := info.Apps[appName]; hook == "" && app != nil && app.IsService() {
+
+	// Do not track apps that don't wish to be tracked.
+	if app := info.Apps[appName]; hook == "" && app.Untracked {
+		needsTracking = false
```

Comment:
The net result is that the app will not be in a cgroup, which means that if the app uses devices that need to be allowed in a device cgroup, on a v2 system we'll not be able to install eBPF filters because it won't be a in a snap specific cgroup.

Perhaps a better way would be do have a check in refresh-app-awareness code to exclude the apps that do not need to stop the refresh?

Reply:
Hmmm, interesting.

Yes, I think that _may_ be better way to do it.

Reply:
I've gone ahead with `RefreshAllowed`, as a new field next to `RefreshMode`, checked in the overlord.

Reply:
Have a look @bboozzoo